packages/fxa-auth-server/lib/routes/session.js

```diff
@@ -548,18 +548,86 @@ module.exports = function (
         const options = {
           title: localizedStrings[titleFtlId],
           body: localizedStrings[bodyFtlId],
-          url,
         };
 
+        const params = new URLSearchParams({
+          tokenVerificationId,
+          code,
+          uid,
+          email,
+          ip: request.app.clientAddress,
+        });
+        const url = `${confirmUrl}?${params.toString()}`;
         try {
-          await push.notifyVerifyLoginRequest(uid, filteredDevices, options);
+          await push.notifyVerifyLoginRequest(uid, filteredDevices, {
+            ...options,
+            url,
+          });
         } catch (err) {
-          log.error('Session.verify.send_push', {
+          log.error('Session.send_push', {
             uid: uid,
             error: err,
           });
         }
 
+        return {};
+      },
+    },
+    {
+      method: 'POST',
+      path: '/session/verify_push',
+      options: {
```

Comment:
It might make sense to have this endpoint be sessionToken authenicated as well. It should be the same user, so in theory we have access to uid, email, etc.